### DISCUSSION TITLE
Find the relation between a program/station and the systemdata.

### DISCUSSION THREAD

**Discussion by fleibede** [dotnetprojects/DotNetSiemensPLCToolBoxLibrary#164] [Q&A]
This is a question or request rather than an issue.

Is there any way to find out which SDB folder that belongs to each CPU?
This is mostly relevant if you have a project with multiple stations.

Through the hOmSave7/s7hk31ax/HRELATI1.dbf you get the CPUFolders and can link it to the program folders.

The compiled system-datablocks are stored in the sdb/"ID"/ folder, and a list of the folders' ID-number can be found in sdb/S0112001.dbf (hex values). How can this ID be linked to the CPUfolder/program-folder structure? 

I have been looking for quite a while to find a tangible connection, but so far been out of luck.
I appreciate any suggestions.

One possible solution that is not ideal, is to search through all .PG files in each sdb folder for the MLFB of the station, and identify the relation through that. It appears the CPU MLFB is present as ASCII in at least one of the .PG files for most projects. The MLFB can be found via this fd41ede commit that i will create a PR with soon.

**Comment by jogibear9988** [comment]
maybe you could ask Thomas_V2.1 from sps-forum.de
He knows many of such details.

From him I also got this:
https://github.com/dotnetprojects/DotNetSiemensPLCToolBoxLibrary/blob/c593752139060e89c9ce6ead666b2a0acf745d4d/LibNoDaveConnectionLibrary/Projectfiles/Step7ProjectV5.cs#L1054
